From f7471886cd3ff5f297af5ca78256d461ce67d78c Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Thu, 27 Jan 2022 10:26:09 +0100
Subject: [PATCH 0148/1695] clk: stm32mp13: removal of LSE clock on RNG clock
 switch in RCC

RNG clock switch becomes 'reserved' on cut1.1

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 1eee7f7dac1ee1ed34b46b04ee3a8b54dab062fb clk: stm32mp13: removal of LSE clock on RNG clock switch in RCC]
Change-Id: Ib5c46241b838bf0f835da4d835ac3da1cbac4587
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
---
 core/drivers/clk/clk-stm32mp13.c                  | 2 +-
 core/include/dt-bindings/clock/stm32mp13-clksrc.h | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/core/drivers/clk/clk-stm32mp13.c b/core/drivers/clk/clk-stm32mp13.c
index 36a36f0aa..bdadcadbd 100644
--- a/core/drivers/clk/clk-stm32mp13.c
+++ b/core/drivers/clk/clk-stm32mp13.c
@@ -2221,7 +2221,7 @@ static STM32_KCLK(ck_saes_k, 4,
 		  0, GATE_SAES, MUX_SAES);
 
 static STM32_KCLK(ck_rng1_k, 4,
-		  PARENT(&ck_csi, &ck_pll4r, &ck_lse, &ck_lsi),
+		  PARENT(&ck_csi, &ck_pll4r, &ck_off, &ck_lsi),
 		  0, GATE_RNG1, MUX_RNG1);
 
 static STM32_KCLK(ck_sdmmc1_k, 4,
diff --git a/core/include/dt-bindings/clock/stm32mp13-clksrc.h b/core/include/dt-bindings/clock/stm32mp13-clksrc.h
index f1f0d8775..43b51b12c 100644
--- a/core/include/dt-bindings/clock/stm32mp13-clksrc.h
+++ b/core/include/dt-bindings/clock/stm32mp13-clksrc.h
@@ -355,7 +355,7 @@
 
 #define CLK_RNG1_CSI		CLKSRC(MUX_RNG1, 0)
 #define CLK_RNG1_PLL4R		CLKSRC(MUX_RNG1, 1)
-#define CLK_RNG1_LSE		CLKSRC(MUX_RNG1, 2)
+/* WARNING: POSITION 2 OF RNG1 MUX IS RESERVED */
 #define CLK_RNG1_LSI		CLKSRC(MUX_RNG1, 3)
 
 #define CLK_STGEN_HSI		CLKSRC(MUX_STGEN, 0)
-- 
2.47.2

